### ISSUE TITLE
About using this code

### ISSUE THREAD

**Issue by ghost** [strawberry-jam-cabal/gpt-mtg#2]
Hello and sorry to bother you,
I came across your code which has come very very handy for the work I'm trying to do. At the moment it does run, but I don't know if it works properly. I have some questions, forgive me for I'm running this while I have absolutely no idea of what I'm doing.
I run the code like this
`python dcgan.py train "\directory" model1`
-Every row takes pretty long to be generated (about 30 sec). I mean this stuff
`33 [D loss: 0.038717, acc: 99.22%] [G loss: 0.028870]`
Is it normal it takes that long? Does that mean it's learning or...? Also, the parameter acc is very high in almost all rows. Is that good or bad? 
I'm using a folder with roughly 200 images, I'm just testing.

-At the moment, only one image has been generated, called model1_0.png. It's a 5x5 grid of smaller squared pics. When are the other samples generated? What if I don't want this kind of grid, just a single big picture?

I hope my requests make some sense, thank you in advance

**Comment by nbertagnolli** [comment]
no apologies needed : )

> Every row takes pretty long to be generated (about 30 sec). I mean this stuff

What type of computer are you running it on?  If you're not running it on a GPU it's going to be pretty slow.  On my 1080TI it takes a few seconds per line.  30 sounds like you might be running it on a CPU?

> Does that mean it's learning or...?

Each line represents some number of learning steps.  The "D loss" is the discriminator loss basically how good is the discriminator at determining fakes from reals the lower the better.  the "G loss" is how well the generator is at fooling the discriminator the lower the better.  That looks like the model is doing pretty well which in the GAN case can be good or bad.  It looks like it's learning alright but 200 images isn't very many so the generator and discriminator have probably just overfit your data.

> -At the moment, only one image has been generated, called model1_0.png.

You can adjust how often it saves these images by setting --save-interval.  It will print out one of these grids every save-interval number of iterations.

To get a single big picture you need to run the generator model only on some 100 dimensional random noise the code is roughly:

`generator = load_model(load_generator_from)
noise = np.random.normal(0, 1, (batch_size, latent_dim))
gen_imgs = generator.predict(noise)
`

let me know if you have any other questions : )
